src/include/distributed/multi_physical_planner.h

```diff
@@ -76,7 +76,8 @@ typedef enum
 	SHARD_FETCH_TASK = 4,
 	MAP_OUTPUT_FETCH_TASK = 5,
 	MERGE_FETCH_TASK = 6,
-	MODIFY_TASK = 7
+	MODIFY_TASK = 7,
+	ROUTER_TASK = 8
```

Comment:
@mtuncer, as we talked in person, we can remove `MODIFY_TASK` and use only `ROUTER_TASK` since there is not much difference among them. We only need to consider couple of small points such as: 
* `LockShardDistributionMetadata()` call on modify task
* Error message decision on `prunedShardCount != 1`

A related change would be to remove `RouterSelectTask()` and `RouterModifyTask()` and merge them into `RouterTask()`. We would need 1 or 2 extra if statements, which is a fair trade-off (I think).